include/zephyr/net/wifi_mgmt.h

```diff
@@ -198,6 +198,9 @@ struct wifi_scan_params {
 	/** Passive scan dwell time (in ms) on a channel.
 	 */
 	uint16_t dwell_time_passive;
+	/** Array of SSID strings to scan.
+	 */
+	char ssids[CONFIG_WIFI_MGMT_SCAN_SSID_FILT_MAX][WIFI_SSID_MAX_LEN];
```

Comment:
better to protect this with `ifdef` of `if` as by default `WIFI_MGMT_SCAN_SSID_FILT` is disabled.

Reply:
`WIFI_MGMT_SCAN_SSID_FILT` is not actually disabling the feature as such and just provides an option to statically specify a SSID filter. However I see your point and have reduced the default value of `CONFIG_WIFI_MGMT_SCAN_SSID_FILT_MAX` to cut down on consumed memory in the default case.